### PR TITLE
Add documentation and remove dead code.

### PR DESCRIPTION
**Pull request by AureliaDolo**

### CODE REVIEW

**Review by AurelienFT — DISMISSED** [review]
Good documentation !

**Comment by AurelienFT** [comment]
What is the rule about capital letter at the start of a documentation line ?

**Comment by AureliaDolo** [comment]
> What is the rule about capital letter at the start of a documentation line ?

maybe we can decide something here #2287 ? 

I have to admit that I am not consistent in formatting.

Once @yvan-sraka tried to format everything

File: massa-consensus-worker/src/consensus_worker.rs
Comment (adrien-zinger) [inline massa-consensus-worker/src/consensus_worker.rs:1193]:
```suggestion
    /// Processing of final blocks, pruning and producing endorsement.
    /// Called at each block tick and each time protocol signal a new block or
    /// a new block header.
```

File: massa-consensus-worker/src/tests/scenario_block_creation.rs
Comment (adrien-zinger) [inline massa-consensus-worker/src/tests/scenario_block_creation.rs:103]: Did you mean ... ?
```suggestion
/// * block at slot( period per cycle + 2, 1)
```

File: massa-client/src/cmds.rs
Comment (AureliaDolo) [reply massa-client/src/cmds.rs:28]: I tried to keep commands that go through private and public api separated and also wallet specific commands last. But really it's just a mess and as there is no convention on naming, alphabetical order may be confusing

Maybe some spec, or some discussion on what we want to do with the client would be useful @damip @Kryptopaid (the russian one looks shinny)

File: massa-consensus-worker/src/tests/scenario_block_creation.rs
Comment (AureliaDolo) [reply massa-consensus-worker/src/tests/scenario_block_creation.rs:103]: I don't remember x)
I just copy pasted the scenario as described in the issue
Maybe it's an issue with the test ?

**Comment by AureliaDolo** [comment]
I think it would be nice to merge this quite quickly to avoid to much conflicts

File: massa-client/src/cmds.rs
Comment (AurelienFT) [inline massa-client/src/cmds.rs:212]: We must sort this out before merging. We can have this type of comment in our codebase.

File: massa-client/src/cmds.rs
Comment (AurelienFT) [inline massa-client/src/cmds.rs:239]:
```suggestion
    /// Reorganize everything into an extended wallet
```

File: massa-models/src/api.rs
Comment (AurelienFT) [inline massa-models/src/api.rs:173]: What "eh" means ?

File: massa-models/src/composite.rs
Comment (AurelienFT) [inline massa-models/src/composite.rs:29]: Use the link method : https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html

**Review by AurelienFT — COMMENTED** [review]
In a more general way, I still don't understand your personal capital rule. Sometimes you have it on fields sometimes not. A lot of time the first line of the doc don't have it but the second one does it's really strange. I understand that there is no global rule but in this context I think the language and uniformity rule should apply.

File: massa-models/src/api.rs
Comment (AureliaDolo) [reply massa-models/src/api.rs:89]: I don't understand your comment

Comment (AurelienFT) [reply massa-models/src/api.rs:89]: Your comment is useful but maybe I missed it. It's that a problem that there is no check or it's just an information ?

Comment (AureliaDolo) [reply massa-models/src/api.rs:89]: It's just an information. If you look at the various check_xxx and note_xxxx_from_node (and maybe the deserialization ones too) the is a description of the checks that are performed and at which level. 

For example, if you take a block, you'll check the parent count at deserialization level only whereas you'll check if the parents are compatible together only at consensus level.

**Review by yvan-sraka — DISMISSED** [review]

File: massa-models/src/api.rs
Comment (AurelienFT) [reply massa-models/src/api.rs:89]: Ok thanks for clarifications :) !

**Review by yvan-sraka — DISMISSED** [review]

**Review by AurelienFT — COMMENTED** [review]
Reviewed half of the files. Still have comments open on my previous review also.

File: massa-network-exports/src/establisher.rs
Comment (AurelienFT) [inline massa-network-exports/src/establisher.rs:6]: I think it should be more explicit. Same for write half.

File: massa-network-exports/src/establisher.rs
Comment (AurelienFT) [inline massa-network-exports/src/establisher.rs:24]: Same as above

File: massa-network-exports/src/test_exports/mock_establisher.rs
Comment (AurelienFT) [inline massa-network-exports/src/test_exports/mock_establisher.rs:13]: I think it should be more explicit. Same for write half.

File: massa-api/src/lib.rs
Comment (AureliaDolo) [reply massa-api/src/lib.rs:40]: Except it has a bad name, from the outside, we don't send commands, we just call consensus command sender's methods

File: massa-bootstrap/src/lib.rs
Comment (AureliaDolo) [reply massa-bootstrap/src/lib.rs:4]: Me neither x), I'll take a look

Comment (AureliaDolo) [reply massa-bootstrap/src/lib.rs:4]: It was a typo: is instead of if

**Comment by AureliaDolo** [comment]
@AurelienFT for the read/write half, I really don't know what to add

File: massa-bootstrap/src/lib.rs
Comment (AureliaDolo) [reply massa-bootstrap/src/lib.rs]: And I added a comma for more clarity

**Comment by AurelienFT** [comment]
> @AurelienFT for the read/write half, I really don't know what to add

On my mind when we say "half" it's half of something it can't just be an half. I think something like "read half of the socket" or whatever it is.

File: massa-api/src/lib.rs
Comment (AurelienFT) [reply massa-api/src/lib.rs:40]: Mhhh ok I think it can be explicited by saying that his methods will interact directly with the consensus worker.

**Review by AurelienFT — COMMENTED** [review]
Very nice documentation ! Still some points to improve on consistency etc but need to be discussed before. I don't approve because we shouldn't merge this before the storage but will make it after.

**Comment by AureliaDolo** [comment]
> Very nice documentation ! Still some points to improve on consistency etc but need to be discussed before. I don't approve because we shouldn't merge this before the storage but will make it after.

Storage is in the bors :)

**Comment by AurelienFT** [comment]
> Storage is in the bors :)

After conflict resolution I will approve

**Comment by AureliaDolo** [comment]
@adrien-zinger @gterzian I squashed all my commits after merging with main.

**Review by AurelienFT — APPROVED** [review]

**Review by yvan-sraka — APPROVED** [review]

**Review by adrien-zinger — APPROVED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Pull request merged**
